### ISSUE TITLE
tcp: Idling connections as SYN-RCV

### ISSUE THREAD

**Issue by AndreasAakesson** [hioa-cs/IncludeOS#542]
Sometimes retransmission from client causes connections to get stuck in SYN-RCV:
```
LISTENING SOCKETS:
10.0.0.42:8081 0.0.0.0:0 (LISTENING)

CONNECTIONS:
Proto	Recv	Send	In	Out	Local			Remote			State
tcp4	 	 	 	 	10.0.0.42:8081		10.0.0.1:39182		SYN-RCV
tcp4	 	 	 	 	10.0.0.42:8081		10.0.0.1:39184		SYN-RCV
tcp4	 	 	 	 	10.0.0.42:8081		10.0.0.1:39186		SYN-RCV
tcp4	 	 	 	 	10.0.0.42:8081		10.0.0.1:39188		SYN-RCV
tcp4	 	 	 	 	10.0.0.42:8081		10.0.0.1:39244		ESTABLISHED
tcp4	 	 	 	 	10.0.0.42:8081		10.0.0.1:39252		ESTABLISHED
```

Not really sure why yet..

**Comment by AndreasAakesson** [comment]
It's probably because the SYN/ACK packet don't get retransmitted. Implementation currently only retransmit packets with payload.

**Comment by AndreasAakesson** [comment]
Fixed in #540 => 78588b0
